testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy

```diff
@@ -395,6 +401,58 @@ abstract class HttpClientTest extends AgentInstrumentationSpecification {
     method = "HEAD"
   }
 
+  /**
+   * This test fires a large number of concurrent requests.
+   * Each request first hits a HTTP server and then makes another client request.
+   * The goal of this test is to verify that in highly concurrent environment our instrumentations
+   * for http clients (especially inherently concurrent ones, such as Netty or Reactor) correctly
+   * propagate trace context.
+   */
+  def "causality test"() {
+    setup:
+    assumeTrue(testCausality())
+    int count = 50
+    def method = 'GET'
+    def url = server.address.resolve("/success")
+
+    def latch = new CountDownLatch(1)
+
+    def pool = Executors.newFixedThreadPool(4)
+
+    when:
+    count.times { index ->
+      def job = {
+        latch.await()
+        runUnderTrace("Parent span " + index) {
+          Span.current().setAttribute("traceAge", index)
```

Comment:
`traceAge` made me think date, maybe `traceId`? ok, no, that's probably more confusing, maybe `testId`? or `testBaggage` 😁? or leave as is totally good too

Reply:
I will leave it as it is for now, I don't think any other proposed name is much better :(

Reply:
oh, i had a better idea for name:
* attribute name: `test.request.id`
* header name: `test-request-id` (if we make it all lowercase then don't need special case for google-http-client below)